Source/WebCore/svg/SVGComponentTransferFunctionElement.h

```diff
@@ -107,4 +109,31 @@ class SVGComponentTransferFunctionElement : public SVGElement {
     Ref<SVGAnimatedNumber> m_offset { SVGAnimatedNumber::create(this) };
 };
 
+inline FEComponentTransfer::Channel SVGComponentTransferFunctionElement::channel() const
+{
+    using namespace WebCore::ElementNames::SVG;
+    using Channel = FEComponentTransfer::Channel;
+
+    switch (elementName()) {
+    case feFuncR: return Channel::Red;
+    case feFuncG: return Channel::Green;
+    case feFuncB: return Channel::Blue;
+    case feFuncA: return Channel::Alpha;
+    default:
+        ASSERT_NOT_REACHED();
+        return Channel::Red;
+    }
+}
+
 } // namespace WebCore
+
+SPECIALIZE_TYPE_TRAITS_BEGIN(WebCore::SVGComponentTransferFunctionElement)
+    static bool isType(const WebCore::Element& element)
+    {
+        using namespace WebCore::ElementNames::SVG;
+        auto name = element.elementName();
+        return name == feFuncR || name == feFuncG || name == feFuncB || name == feFuncA;
```

Comment:
Can not this function be expressed using the superclasses traits instead:
```
return is<SVGFEFuncRElement> || is<SVGFEFuncGElement> || ...
```

Reply:
Oh, yes it can. That looks better.